lib/route/cls/ematch.c

```diff
@@ -289,6 +289,65 @@ void rtnl_ematch_tree_free(struct rtnl_ematch_tree *tree)
 	free(tree);
 }
 
+static int clone_ematch_list(struct nl_list_head *dst, struct nl_list_head *src)
+{
+	struct rtnl_ematch *new = NULL, *pos = NULL;
+
+	nl_list_for_each_entry(pos, src, e_list) {
+		new = rtnl_ematch_alloc();
+		if (!new)
+			goto nomem;
+
+		new->e_id      = pos->e_id;
+		new->e_kind    = pos->e_kind;
+		new->e_flags   = pos->e_flags;
+		new->e_index   = pos->e_index;
+		new->e_datalen = pos->e_datalen;
+
+		if (pos->e_ops) {
+			if (rtnl_ematch_set_ops(new, pos->e_ops))
+				goto nomem;
+		}
+
+		if (!nl_list_empty(&pos->e_childs)) {
+			if (clone_ematch_list(&new->e_childs, &pos->e_childs))
```

Comment:
Check for failure with `< 0`.

Reply:
Because, errors should be signaled by negative integers and the preferred style for checking whether an integer return value signals an error is `if (r < 0)`. Arguably, if you look at `clone_ematch_list` it never returns a positive value. But one shouldn't need to look at `clone_ematch_list` to verify that.

Reply:
ah, if you find places where this is not honoured, that is a bug. Likely it's not a serious bug, only violating the style. Those places should be fixed, but well. Let's get it at least that way for new code.